ml-search-server/api/controllers/searchController.js

```diff
@@ -1,37 +1,39 @@
-            },
-            {
-                url: '/D/POC/ml-search-poc/2.json',
-                text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.'
-            },
-            {
-                url: '/D/POC/ml-search-poc/3.json',
-                text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.'
-            },
-            {
-                url: '/D/POC/ml-search-poc/4.json',
-                text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.'
-            },
-            {
-                url: '/D/POC/ml-search-poc/5.json',
-                text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.'
-            },
-            {
-                url: '/D/POC/ml-search-poc/6.json',
-                text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.'
-            },
-            {
-                url: '/D/POC/ml-search-poc/7.json',
-                text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.'
-            }
-        ]
+    var qb = marklogic.queryBuilder;
+    const pageNumber = req.query.page_no;
+    //const pageSize = 10;
+    const pageSize = req.query.page_size;
+    const start = (pageNumber - 1) * pageSize;
+    const query = db.documents.query(
+        qb.where(qb.byExample({ tags: req.query.text }))
+    )
+    query.result(function(documents) {
+
+        const end = start + pageSize;
+
+        res.header("Access-Control-Allow-Origin", "*");
+        res.header("Access-Control-Allow-Headers", "X-Requested-With");
+        const data = _.slice(documents, start, start + pageSize).map((document) => {
+            return { url: document.uri, text: document.content.text.substring(0, 100) };
+        })
+        res.send({ totalCounts: documents.length, data: data });
```

Comment:
add a TODO comment stating, this query needs to optimise to get the document of the current page only